SilKit/source/services/logging/Logger.cpp

```diff
@@ -118,32 +314,124 @@ Logger::Logger(const std::string& participantName, Config::Logging config)
+
+            if (sink.format == Config::Sink::Format::Json && sink.type != Config::Sink::Type::Remote)
+            {
+                using spdlog::details::make_unique; // for pre c++14
+                auto formatter = make_unique<spdlog::pattern_formatter>();
+                formatter->add_flag<epoch_formatter_flag>('E').set_pattern("[%E] [%^%l%$] %v");
+                formatter->set_pattern(jsonpattern);
+                stdoutSink->set_formatter(std::move(formatter));
+                stdoutSink->set_level(log_level);
+                _loggerJson->sinks().emplace_back(std::move(stdoutSink));
+            }
+            else if(sink.type != Config::Sink::Type::Remote)
+            {
+                stdoutSink->set_level(log_level);
+                _loggerSimple->sinks().emplace_back(std::move(stdoutSink));
+            }
             break;
         }
         case Config::Sink::Type::File:
         {
-            //
             auto filename = fmt::format("{}_{:%FT%H-%M-%S}.txt", sink.logName, tmBuffer);
             auto fileSink = std::make_shared<spdlog::sinks::basic_file_sink_mt>(filename);
-            fileSink->set_level(log_level);
-            _logger->sinks().push_back(fileSink);
+
+            if (sink.format == Config::Sink::Format::Json)
+            {   
+                using spdlog::details::make_unique; // for pre c++14
+                auto formatter = make_unique<spdlog::pattern_formatter>();
+                formatter->add_flag<epoch_formatter_flag>('E').set_pattern("[%E] [%^%l%$] %v");
+                formatter->set_pattern(jsonpattern);
+                fileSink->set_formatter(std::move(formatter));
+                fileSink->set_level(log_level);
+                _loggerJson->sinks().push_back(fileSink);
+            }
+            else
+            {
+                fileSink->set_level(log_level);
+                _loggerSimple->sinks().push_back(fileSink);
+            }
         }
         }
     }
+    if (nullptr != _loggerSimple)
+    {
+        _loggerSimple->flush_on(to_spdlog(_config.flushLevel));
+    }
+    if (nullptr != _loggerJson)
+    {
+        _loggerJson->flush_on(to_spdlog(_config.flushLevel));
+    }
+}
 
-    _logger->flush_on(to_spdlog(_config.flushLevel));
+void Logger::ProcessLoggerMessage(const LoggerMessage& msg)
+{
+    const auto now = log_clock::now();
+    if (nullptr != _loggerJson)
+    {
+        JsonLogMessage myJsonMsg{msg};
+        _loggerJson->log(now, spdlog::source_loc{}, to_spdlog(msg.GetLevel()), fmt::format("{}", myJsonMsg));
+    }
+
+    if (nullptr != _loggerSimple)
+    {
+        SimpleLogMessage myMsg{msg};
+        _loggerSimple->log(now, spdlog::source_loc{}, to_spdlog(msg.GetLevel()), fmt::format("{}", myMsg));
+    }
+    if (nullptr != _loggerRemote)
+    {
+        _loggerRemote->Log(now, msg);
+    }
 }
 
+void Logger::LogReceivedMsg(const LogMsg& msg)
+{
+    LoggerMessage loggerMsg{this, msg};
```

Comment:
For received messages you have to create a `spdlog::details::log_msg` from the `LogMsg`, similar to how the current code on [main](https://github.com/vectorgrp/sil-kit/blob/main/SilKit/source/services/logging/Logger.cpp#L204) does it. You can adapt the [`to_spdlog`](https://github.com/vectorgrp/sil-kit/blob/main/SilKit/source/services/logging/SpdlogTypeConversion.hpp#L114) conversions for that.

Reply:
For constructing the `log_msg` use the constructor where the first parameter is a `log_clock::time_point` (see [here](https://github.com/gabime/spdlog/blob/v1.x/include/spdlog/details/log_msg.h#L13)).